src/solver/variable/economy/STStorageCashFlowByCluster.h

```diff
@@ -0,0 +1,304 @@
+    //! VCard
+    typedef VCardSTstorageCashFlowByCluster VCardType;
+    //! Ancestor
+    typedef Variable::IVariable<STstorageCashFlowByCluster<NextT>, NextT, VCardType> AncestorType;
+
+    //! List of expected results
+    typedef typename VCardType::ResultsType ResultsType;
+
+    typedef VariableAccessor<ResultsType, VCardType::columnCount> VariableAccessorType;
+
+    enum
+    {
+        //! How many items have we got
+        count = 1 + NextT::count,
+    };
+
+    template<int CDataLevel, int CFile>
+    struct Statistics
+    {
+        enum
+        {
+            count
+            = ((VCardType::categoryDataLevel & CDataLevel && VCardType::categoryFileLevel & CFile)
+                 ? (NextType::template Statistics<CDataLevel, CFile>::count
+                    + VCardType::columnCount * ResultsType::count)
+                 : NextType::template Statistics<CDataLevel, CFile>::count),
+        };
+    };
+
+public:
+    STstorageCashFlowByCluster() = default;
+
+    ~STstorageCashFlowByCluster()
+    {
+        for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+            delete[] pValuesForTheCurrentYear[numSpace];
+        delete[] pValuesForTheCurrentYear;
+    }
+
+    void initializeFromArea(Data::Study* study, Data::Area* area)
+    {
+        // Get the number of years in parallel
+        pNbYearsParallel = study->maxNbYearsInParallel;
+        pValuesForTheCurrentYear = new VCardType::IntermediateValuesBaseType[pNbYearsParallel];
+
+        // Get the area
+        nbClusters_ = area->shortTermStorage.count();
+        if (nbClusters_)
+        {
+            AncestorType::pResults.resize(nbClusters_);
+
+            for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+                pValuesForTheCurrentYear[numSpace]
+                  = new VCardType::IntermediateValuesDeepType[nbClusters_];
+
+            for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+                for (unsigned int i = 0; i != nbClusters_; ++i)
+                    pValuesForTheCurrentYear[numSpace][i].initializeFromStudy(*study);
+
+            for (unsigned int i = 0; i != nbClusters_; ++i)
+            {
+                AncestorType::pResults[i].initializeFromStudy(*study);
+                AncestorType::pResults[i].reset();
+            }
+        }
+        else
+        {
+            for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+            {
+                pValuesForTheCurrentYear[numSpace] = nullptr;
+            }
+
+            AncestorType::pResults.clear();
+        }
+        // Next
+        NextType::initializeFromArea(study, area);
+    }
+
+    size_t getMaxNumberColumns() const
+    {
+        return nbClusters_ * ResultsType::count;
+    }
+
+    void yearBegin(unsigned int year, unsigned int numSpace)
+    {
+        // Reset the values for the current year
+        for (unsigned int i = 0; i != nbClusters_; ++i)
```

Comment:
`for (unsigned int i = 0; i != nbClusters_; ++i)` : **i** for a cluster index ? **clusterIndex** is better, or at least **c**